std/traits.d

```diff
@@ -8087,3 +8089,97 @@ enum isCopyable(S) = is(typeof(
     static assert(isCopyable!int);
     static assert(isCopyable!(int[]));
 }
+
+/**
+ * Determines the type of the argument.
+ *
+ * Returns:
+ *   T[0] if it's a type, typeof(T[0]) if it's a value, void if it's a template.
+ *
+ */
+template TypeOf(T...)
```

Comment:
I would set this to `private` and don't expose via ddoc. At least until you have convinced us that it's useful on its own.